pkg/templates/templates.go

```diff
@@ -102,7 +110,7 @@ func ResolveTemplate(tmplMap interface{}) (interface{}, error) {
 	}
 
 	var buf strings.Builder
-	err = tmpl.Execute(&buf, "")
+	err = tmpl.Execute(&buf, tmplCtx)
```

Comment:
doesnt this change break config policy controller ?   you need to provide an option where caller is not required to set it. managedcluster dont set it

Reply:
It wouldn't break it since `config-policy-controller` is pinned to the commit currently in the main branch. When it's ready to absorb this change, it'll have to upgrade its `go.mod` entry and change the way it calls this code.

Go doesn't have optional parameters. So perhaps the idea I had in the previous comment of making a struct containing the public API methods would work.

Reply:
Consider a user of this util pkg , who wants to resolve templates in his yaml ( not necessarily policy yaml but any yaml).
ClusterId or ClusterName may not be something that make sense in that user's case.

In such a case , it appears  user is required to pass a context with unnecessary variables set to empty. Would be nice if the user doesnt have to set it at all if its not needed for their case.


but  its alright for now I guess as its only being used by the policy components currently

Reply:
> Go doesn't have optional parameters

True, although you can get something similar with variadic functions/arguments